### ISSUE TITLE
Fix `wai-aria` rule's validation

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#673]
- Markuplint version: `3.0.1`
- Parser lang: HTML
- Node.js version: `16.16.0`
- OS: Windows 11

## Describe the bug

`wai-aria` rule warns `role="radiogroup"` that has `role="radio"` as grandchildren elements.

## Code Example

```html
<ul role="radiogroup">
    <li role="presentation"><button role="radio" aria-checked="true">A</button></li>
    <li role="presentation"><button role="radio" aria-checked="false">B</button></li>
    <li role="presentation"><button role="radio" aria-checked="false">C</button></li>
</ul>
```

## Steps To Reproduce

1. Lint the code above.
2. Markuplint shows message below:

```
ロール「radiogroup」にはロール「radio」が必要です
```

## The current behavior

`wai-aria` warns the code.

## The expected behavior

`wai-aria` doesn't warn the code.

(Sorry if I'm wrong.)

**Comment by YusukeHirao** [comment]
## Investigation Report

I investigated this issue by writing verification tests against both ARIA 1.2 and 1.3. See PR #3233 for regression tests.

### Root cause: Implementation bug (now fixed)

The reporter was correct. The `requiredOwnedElements` check in `required-owned-elements.ts` only examined **direct child elements** and did not treat `presentation`/`none` roles as transparent wrappers. This caused a false positive: `radiogroup` could not "see through" the `li[role=presentation]` to find the `button[role=radio]` grandchild.

### Fix

This was fixed in February 2026 during the ARIA 1.3 generic role transparency implementation. While the work was done as part of the ARIA 1.3 effort, the fix for `presentation`/`none` transparency applies to **all ARIA versions** (1.1, 1.2, and 1.3):

- `isTransparentForOwnership()` was added to `ml-spec`, defining `presentation` and `none` as **always** transparent for ownership checks regardless of ARIA version (and additionally `generic` as transparent in ARIA 1.3 only)
- `classifyChildren()` was added to recursively skip transparent wrappers when checking required owned elements

Before this fix (prior to Feb 2026), the bug existed in all ARIA versions. After the fix, it is resolved in all versions.

### Current behavior (verified in both ARIA 1.2 and 1.3)

All of the following produce **no false positive**:

- `radiogroup > li[role=presentation] > button[role=radio]` — no requiredOwnedElements violation
- `radiogroup > li[role=none] > button[role=radio]` — no requiredOwnedElements violation
- `radiogroup > div[role=radio]` (direct child) — no violation

Regression tests added in PR #3233.

**Comment by YusukeHirao** [comment]
Closing: Fixed in February 2026. presentation/none transparency now works in all ARIA versions. See investigation report above.
